src/routes/article.ts

```diff
@@ -0,0 +1,31 @@
+import clownface from 'clownface';
+import createHttpError from 'http-errors';
+import { constants } from 'http2';
+import { Next } from 'koa';
+import { addAll } from 'rdf-dataset-ext';
+import { AppContext, AppMiddleware } from '../app';
+
+export default (): AppMiddleware => (
+  async ({
+    dataFactory: { namedNode }, articles, path, response,
+  }: AppContext, next: Next): Promise<void> => {
+    if (!response.status || response.status === constants.HTTP_STATUS_NOT_FOUND) {
+      const articleNamedNode = namedNode(path);
+      const graph = clownface({
+        dataset: response.dataset,
+        term: articleNamedNode,
+      });
+      try {
+        await articles.get(articleNamedNode).then((article) => {
+          addAll(graph.dataset, article);
+        });
+      } catch (error) {
+        throw new createHttpError.NotFound(error.message);
```

Comment:
The `try`/`catch` block shouldn't wrap other code (only the `get`), and it should check for the error type (`ArticleNotFound`).  At the moment _any_ error, including those from Clownface are turned into `NotFound`.